proposals/0000-initial-extension-categorization.md

```diff
@@ -0,0 +1,234 @@
+| DeepSubsumption          | Legacy         | Stable                  |
+| NondecreasingIndentation | Legacy         | Stable                  |
+| NPlusKPatterns           | Legacy         | Stable                  |
+| StarIsType               | Legacy         | Stable                  |
+
+### 2.2.2 Experimental Extensions
+
+The following are extensions that are to be categorized as `Experimental` when enabled, and for the `No` or negative version are categorized as `Stable`. As a reminder, the [categorizations](https://github.com/ghc-proposals/ghc-proposals/blob/master/proposals/0601-extension-lifecycle-framework.md#21-categories) state that `Experimental` extensions "are undergoing active development, or have consequences that make it impossible in practice to avoid breaking changes".
+
+The following table contains the language extension, the proposed categorization for the extension being turned on, and the proposed categorization for the extension being turned off.
+
+| Extension              | Categorization | Negative Categorization |
+|------------------------|----------------|-------------------------|
+| ApplicativeDo          | Experimental   | Stable                  |
+| Arrows                 | Experimental   | Stable                  |
+| GHCForeignImportPrim   | Experimental   | Stable                  |
+| ExtendedLiterals       | Experimental   | Stable                  |
+| ImpredicativeTypes     | Experimental   | Stable                  |
+| JavaScriptFFI          | Experimental   | Stable                  |
+| LinearTypes            | Experimental   | Stable                  |
+| OverloadedRecordUpdate | Experimental   | Stable                  |
+| RequiredTypeArguments  | Experimental   | Stable                  |
+
+### 2.2.3 Deprecated Extensions
+
+The following are extensions that are to be categorized as `Deprecated` when enabled, and for the `No` version are categorized as `Stable`.
+
+| Extension                         | Categorization | Negative Categorization |
+|-----------------------------------|----------------|-------------------------|
+| AlternativeLayoutRule             | Deprecated     | Stable                  |
+| AlternativeLayoutRuleTransitional | Deprecated     | Stable                  |
+| DoRec                             | Deprecated     | Stable                  |
+| NullaryTypeClasses                | Deprecated     | Stable                  |
+| OverlappingInstances              | Deprecated     | Stable                  |
+| ParallelArrays                    | Deprecated     | Stable                  |
+| PolymorphicComponents             | Deprecated     | Stable                  |
+| Rank2Types                        | Deprecated     | Stable                  |
+| RecordPuns                        | Deprecated     | Stable                  |
+| TypeInType                        | Deprecated     | Stable                  |
```

Comment:
Why is the "negative categorization" marked as "stable" for TypeInType? You get the warning either way

```
ghci> :set -XTypeInType
<no location info>: warning: [GHC-53692] [-Wdeprecated-flags]
    -XTypeInType is deprecated: use -XDataKinds and -XPolyKinds instead

ghci> :set -XNoTypeInType
<no location info>: warning: [GHC-53692] [-Wdeprecated-flags]
    -XNoTypeInType is deprecated: use -XNoDataKinds and -XNoPolyKinds instead
```

i.e. the entire extension flag is deprecated.

The same applies to